packages/@aws-cdk/custom-resource-handlers/lib/aws-synthetics-alpha/auto-delete-underlying-resources-handler/index.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-console */
+/* eslint-disable import/no-extraneous-dependencies */
+import { LambdaClient, DeleteFunctionCommand } from '@aws-sdk/client-lambda';
+import { SyntheticsClient, GetCanaryCommand } from '@aws-sdk/client-synthetics';
+import { makeHandler } from '../../nodejs-entrypoint';
+
+const AUTO_DELETE_LAMBDA_TAG = 'aws-cdk:auto-delete-lambda';
+
+const lambda = new LambdaClient({});
+const synthetics = new SyntheticsClient({});
+
+export const handler = makeHandler(autoDeleteHandler);
+
+export async function autoDeleteHandler(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  switch (event.RequestType) {
+    case 'Create':
+      break;
+    case 'Update':
+      return onUpdate(event);
+    case 'Delete':
+      return onDelete(event.ResourceProperties?.CanaryName);
+  }
+};
+
+async function onUpdate(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const updateEvent = event as AWSLambda.CloudFormationCustomResourceUpdateEvent;
+  const oldCanaryName = updateEvent.OldResourceProperties?.CanaryName;
```

Comment:
This might be correct, but it feels more natural to me to get this from `PhysicalResourceId`.

And then the rest of this method is also quite trivial, because you can always just return `newCanaryName` (if it's the same, that's fine!)